src/it/resources/com/google/checkstyle/test/chapter5naming/rule523methodnames/InputMethodName.java

```diff
@@ -22,6 +24,27 @@ void f() {} // violation 'Method name 'f' must match pattern'
 
   void fO() {} // violation 'Method name 'fO' must match pattern'
 
+  @Test
+  void testing_foo() {}
+
+  @Test
+  void testing_Foo() {}
+
+  @Test
+  void testing_fOo() {}
+
+  @Test
+  void testingFoo() {}
+
+  @Test
+  void testingFoo_foo() {}
+
+  @Test
+  void testing_0123() {}
```

Comment:
same as above, after underscore the first letter should not be a number? But I'm not so sure about this as documentation does not mentions anything about using digits.

[5.2.3 Method names](https://google.github.io/styleguide/javaguide.html#s5.2.3-method-names)

Reply:
They say nothing about numbers. Could you check if there are rules in other sections regarding the usage of numbers in names?

Reply:
I found this from [5.1 Rules common to all identifiers](https://google.github.io/styleguide/javaguide.html#s5.1-identifier-names):

> Identifiers use only ASCII letters and digits, and, in a small number of cases noted below, underscores

though this section is not about method naming but it does mentions that numbers are allowed.

Reply:
number that starts a word is ok, they have example: `"Turn on 2SV" --->	turnOn2sv`

Reply:
but numbers directly after `_` are ok? Nobody write name likes this, so I'm bit confuse

Reply:
Turn on 2SV --> Turn_on_2SV

> Nobody write name likes this

https://en.wikipedia.org/wiki/Snake_case , it is one of styles.
Java trending more to CamleCase style.